### DISCUSSION TITLE
How can we set retention time on topic level using Pulsar Admin in Java?

### DISCUSSION THREAD

**Discussion by anijain-D11** [apache/pulsar#21761] [Q&A] [answered]
Below is the snippet to create topic in service:

  public void createTopic(
      Long tenantId, Env env, String namespaceName, String topicName, int numPartitions) {
    String tenantName = tenantsService.findById(tenantId).getName();
    getPulsarAdmin(tenantId, env, PulsarRole.ADMIN)
        .topics()
        .createPartitionedTopic(getTopicName(tenantName, namespaceName, topicName), numPartitions);
  }

I can't see any function to set Retention Time on the Topic level. 

The following method is deprecated.
void setRetention(String topic, RetentionPolicies retention) throws PulsarAdminException;

**Comment by alpreu** [comment, marked as answer]
Hi Ani,
`Topic#setRetention` is deprecated, yes. You should use `TopicPolicies#setRetention` instead
